js/digifinex.js

```diff
@@ -0,0 +1,528 @@
+                'TUSD': 'TUSD',
+                'USDC': 'USDC',
+            },
+            'exceptions': {
+                'exact': {
+                },
+                'broad': {
+                },
+            },
+            'precisionMode': DECIMAL_PLACES,
+            'options': {
+                'orderTypes': {
+                },
+            },
+            'apiKey': '15cb6aad284ab8',
+            'secret': '48cad555befb5de238f6688b440a803f05cb6aad2',
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        let response = await this.privateGetTradePairs ();
+        // if (response.code != 0)
+        //     log.info ('[fetchMarkets] error: ' + response.code);
+        // console.log (response);
+        let result = [];
+        let keys = Object.keys (response['data']);
+        for (let i = 0; i < keys.length; i++) {
+            let symbol = keys[i];
+            let id = symbol.toUpperCase ();
+            let pair = id.split ('_');
+            let base = pair[0];
+            let quote = pair[1];
+            let baseId = base;
+            let quoteId = quote;
+            let market = response['data'][symbol];
+            let precision = {
+                'price': market[1],
+                'amount': market[0],
+            };
+            let limits = {
+                'amount': {
+                    'min': market[2],
+                    'max': undefined,
+                },
+                'price': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'cost': {
+                    'min': market[3],
+                    'max': undefined,
+                },
+            };
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': true,
+                'precision': precision,
+                'limits': limits,
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    async fetchBalance (params = {}) {
+        let response = await this.privateGetMyposition ();
+        let result = { 'info': response };
+        let free = response['free'];
+        let frozen = response['frozen'];
+        let keysFree = Object.keys (free);
+        let keysFrozen = Object.keys (frozen);
+        let keys = this.arrayConcat (keysFree, keysFrozen);
+        for (let i = 0; i < keys.length; i++) {
+            let uppercase = keys[i].toUpperCase ();
+            let account = this.account ();
+            if (this.inArray (keys[i], keysFree)) {
+                account['free'] = free[keys[i]];
+            }
+            if (this.inArray (keys[i], keysFrozen)) {
+                account['used'] = frozen[keys[i]];
+            }
+            account['total'] = account['free'] + account['used'];
```

Comment:
We need to use `this.sum` in this line ↑ for reasons explained here: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings